### PR TITLE
Add variables and variable_file in terraform module

### LINKED ISSUE #36837: Terraform module doesn't use variables dictionary
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
- terraform

##### ANSIBLE VERSION
```
ansible 2.5.0b1
```

##### CONFIGURATION
Just main.yml

##### OS / ENVIRONMENT
WSL, Ubuntu 16.04.3 LTS

##### SUMMARY
I want to try terraform module from ansible 2.5 beta version. Terraform module always validates terraform files before starts applying, but it doesn't include variables specified in terraform module `variables` section. 

##### STEPS TO REPRODUCE
- Terraform variable file:
```variables.tf:
variable "cluster_tag" {
  description = "Tag of Cluster"
}

variable "aws_access_key" {}

variable "aws_secret_key" {}

variable "aws_region" {
  description = "Region for AWS"
}
```
- ansible-playbook -c local main.yml

```- hosts: localhost
  tasks:
  - name: Provide infrastructure
    terraform:
      variables:
        aws_access_key: <access_key>
        aws_secret_key: <secret_keky>
        aws_region: eu-west-1
        cluster_tag: ansible-test
      project_path: ./terraform
```

##### EXPECTED RESULTS
Terraform considers specified variables, validate and will run specified actions, if terraform file validate successful.

##### ACTUAL RESULTS
```
ansible-playbook -c local main.yml
 [WARNING]: Unable to parse /etc/ansible/hosts as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAY [localhost] **********************************************************************************************************************************************

TASK [Gathering Facts] ****************************************************************************************************************************************
ok: [localhost]

TASK [Provide infrastructure] ********************************************************************************************************************************$
fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true
}

MSG:
Failed to validate Terraform configuration files:

Error: Required variable not set: aws_region
Error: Required variable not set: aws_access_key
Error: Required variable not set: aws_secret_key
Error: Required variable not set: cluster_tag
```

### PR DESCRIPTION
**Pull request by okgolove**
##### SUMMARY

Fixes #36837
Added variables and variable_file options for validate command.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

terraform

##### ANSIBLE VERSION

```
ansible 2.6.0
  config file = None
  configured module search path = [u'/home/mnaletov/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/mnaletov/.local/share/virtualenvs/ansible-UKLfXmN5/local/lib/python2.7/site-packages/ansible
  executable location = /home/mnaletov/.local/share/virtualenvs/ansible-UKLfXmN5/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

### CODE REVIEW

File: lib/ansible/modules/cloud/misc/terraform.py [hunk 1]
```diff
@@ -209,7 +209,16 @@ def main():
...
 
-    preflight_validation(command[0], project_path)
+    variables_args = []
+    for k, v in variables.items():
+        variables_args.extend([
+            '-var',
+            '{0}={1}'.format(k, v)
+        ])
+    if variables_file:
+        variables_args.append('-var-file', variables_file)
```
Comment (sivel) [inline lib/ansible/modules/cloud/misc/terraform.py]: This likely needs to be:

```
variables_args.extend(['-var-file', variables_file])
```

File: lib/ansible/modules/cloud/misc/terraform.py [hunk 2]
```diff
@@ -136,13 +136,13 @@
 module = None
 
 
-def preflight_validation(bin_path, project_path, variables_file=None, plan_file=None):
+def preflight_validation(bin_path, project_path, variables_args=None, plan_file=None):
```
Comment (ryansb) [inline lib/ansible/modules/cloud/misc/terraform.py:139]: I think this needs to include both the vars args and vars file. Looks like excluding the vars_file in the preflight command was an oversight on my part when implementing this originally.

**Comment by okgolove** [comment]
I'll fix it tomorrow.

File: lib/ansible/modules/cloud/misc/terraform.py (hunk 2 above)
Comment (okgolove) [reply lib/ansible/modules/cloud/misc/terraform.py:139]: But we have already added vars_file option in 219 line:
```python
if variables_file:
        variables_args.append('-var-file', variables_file)
```

Comment (ryansb) [reply lib/ansible/modules/cloud/misc/terraform.py:139]: Ah, good eye.

**Pull request merged**